packages/@aws-cdk/aws-stepfunctions-tasks/README.md

```diff
@@ -326,31 +328,87 @@ Step Functions supports [ECS/Fargate](https://docs.aws.amazon.com/step-functions
+
+The following example runs a job from a task definition on EC2
+
 ```ts
 import * as ecs from '@aws-cdk/aws-ecs';
+import * as tasks from '@aws-cdk/aws-stepfunctions-tasks';
+import * as sfn from '@aws-cdk/aws-stepfunctions';
 
-// See examples in ECS library for initialization of 'cluster' and 'taskDefinition'
+const taskDefinition = new ecs.TaskDefinition(stack, 'TD', {
+  compatibility: ecs.Compatibility.EC2,
+});
 
-new ecs.RunEcsFargateTask({
-  cluster,
-  taskDefinition,
-  containerOverrides: [
-    {
-      containerName: 'TheContainer',
-      environment: [
-        {
-          name: 'CONTAINER_INPUT',
-          value: Data.stringAt('$.valueFromStateData'),
-        }
-      ]
-    }
-  ]
+taskDefinition.addContainer('TheContainer', {
+  image: ecs.ContainerImage.fromRegistry('foo/bar'),
+  memoryLimitMiB: 256,
+});
+
+const runTask = new tasks.EcsEc2RunTask(stack, 'Run', {
+    integrationPattern: sfn.IntegrationPattern.RUN_JOB,
+    cluster,
+    taskDefinition,
+    placementStrategies: [
+      ecs.PlacementStrategy.spreadAcrossInstances(),
+      ecs.PlacementStrategy.packedByCpu(),
+      ecs.PlacementStrategy.randomly(),
+    ],
+    placementConstraints: [
+      ecs.PlacementConstraint.memberOf('blieptuut')
+    ],
+  });
+```
+
+#### Fargate
+
+AWS Fargate is a serverless compute engine for containers that works with Amazon
+Elastic Container Service (ECS). Fargate makes it easy for you to focus on building
+your applications. Fargate removes the need to provision and manage servers, lets you
+specify and pay for resources per application, and improves security through application
+isolation by design. Learn more about [Fargate](https://aws.amazon.com/fargate/)
+
+The Fargate launch type allows you to run your containerized applications without the need
+to provision and manage the backend infrastructure. Just register your task definition and
+Fargate launches the container for you.
+
+The following example runs a job from a task definition on Fargate
+
+```ts
+import * as ecs from '@aws-cdk/aws-ecs';
+import * as tasks from '@aws-cdk/aws-stepfunctions-tasks';
+import * as sfn from '@aws-cdk/aws-stepfunctions';
+
+
+const taskDefinition = new ecs.TaskDefinition(stack, 'TD', {
+  memoryMiB: '512',
+  cpu: '256',
+  compatibility: ecs.Compatibility.FARGATE,
 });
 
-fargateTask.connections.allowToDefaultPort(rdsCluster, 'Read the database');
+taskDefinition.addContainer('TheContainer', {
+  image: ecs.ContainerImage.fromRegistry('foo/bar'),
+  memoryLimitMiB: 256,
+});
 
-new sfn.Task(this, 'CallFargate', {
-  task: fargateTask
+const runTask = new tasks.EcsFargateRunTask(stack, 'RunFargate', {
+  integrationPattern: sfn.IntegrationPattern.RUN_JOB,
+  cluster,
+  taskDefinition,
+  containerOverrides: [{
+    containerName: 'TheContainer',
+    environment: [{ name: 'SOME_KEY', value: sfn.Data.stringAt('$.SomeKey') }],
+  }],
```

Comment:
> the primary reason the overrides are powerful is because they allow selection and definition from state input (i.e. sfn.Data.stringAt(...))
> 
> Some of the Step Functions sample documentation leads into it and I've seen from a few users who have shared code that it's being used. If overrides are not specified, the values established the task definition will be used by default.

I was not asking about the need for `containerOverrides` themselves but the reference to `containerName`.

Is it correct to say, that the container names specified in the overrides must match the containers configured on the task?

If so, should we do something like

```ts
containerOverrides: [{
  container: container,
  environment: ...
}]
```

Reply:
thanks for clarifying. Yes, it's correct to say that an override can only be applied to a container specified in the task definition. changing this to accept `ContainerDefinition` instead of `containerName`